### PR TITLE
Fluent Validation for password

### LINKED ISSUE #400: (Fluent Validation) Add regular expression check for password
In validators for account you need to add some regex validations for password
Method for checking password should be created

### PR DESCRIPTION
**Pull request by TetianaDomanska-dev**
Now we check validation of password due to fluent validation

### CODE REVIEW

File: CharlieBackend.Business/Helpers/PasswordHelper.cs [hunk 1]
```diff
@@ -65,41 +65,9 @@ private static Int32 Next(Int32 minValue, Int32 maxValue)
...
-                allWarnings.Append("Password should contain at least one lower case letter. ");
-            }
-
-            if (Regex.IsMatch(password, @"[#?!@$%^&*-]+"))
-            {
-                allWarnings.Append("Password can't contain any special character apart from underscore symbol. ");
-            }
-
-            return allWarnings.ToString();
+            return Regex.IsMatch(password.ToString(), _pattern);
```
Comment (andy-dendy-repo) [inline CharlieBackend.Business/Helpers/PasswordHelper.cs]:
```suggestion
            return Regex.IsMatch(password, _pattern);
```

**Review by andy-dendy-repo — APPROVED** [review]

File: CharlieBackend.Business/Helpers/PasswordHelper.cs (hunk 1 above)
Comment (TetianaDomanska-dev) [reply CharlieBackend.Business/Helpers/PasswordHelper.cs]: Thanks, fixed

File: CharlieBackend.Api/Validators/AccountDTOValidators/ChangeCurrentPasswordDtoValidator.cs [hunk 2]
```diff
@@ -16,11 +13,15 @@ public ChangeCurrentPasswordDtoValidator()
                 .EmailAddress().WithMessage("Incorrect email")
                 .MaximumLength(50).WithMessage("Email cannot be greateh than {MaxLength} symbols");
```
Comment (gingerbread-monster) [inline CharlieBackend.Api/Validators/AccountDTOValidators/ChangeCurrentPasswordDtoValidator.cs]: I think FluentValidation package provides the default error messages that would be pretty same to these.
Please, remove unnecessary error messages

File: CharlieBackend.Api/Validators/AccountDTOValidators/ChangeCurrentPasswordDtoValidator.cs [hunk 3]
```diff
@@ -16,11 +13,15 @@ public ChangeCurrentPasswordDtoValidator()
...
             RuleFor(x => x.CurrentPassword)
-                .NotEmpty().WithMessage("{PropertyName} is required")
-                .MaximumLength(30).WithMessage("Password cannot be greater than {MaxLength} symbols");
+                .NotEmpty()
+                .MinimumLength(8)
+                .MaximumLength(30)
```
Comment (gingerbread-monster) [inline CharlieBackend.Api/Validators/AccountDTOValidators/ChangeCurrentPasswordDtoValidator.cs]: Please, remove the recurring magic numbers

File: CharlieBackend.Api/Validators/AccountDTOValidators/AuthenticationDtoValidator.cs [hunk 4]
```diff
@@ -15,8 +13,10 @@ public AuthenticationDtoValidator()
                 .EmailAddress().WithMessage("Incorrect email")
                 .MaximumLength(50).WithMessage("Email cannot be greateh than {MaxLength} symbols");
             RuleFor(x => x.Password)
-                .NotEmpty().WithMessage("Password cannot be empty")
-                .MaximumLength(65).WithMessage("Password cannot be greater than {MaxLength} symbols");
+                .NotEmpty()
+                .MinimumLength(8)
+                .MaximumLength(30)
+                .Must(PasswordHelper.PasswordValidation).WithMessage("Password must have at least eight characters, at least one uppercase letter, one lowercase letter and one number");
```
Comment (gingerbread-monster) [inline CharlieBackend.Api/Validators/AccountDTOValidators/AuthenticationDtoValidator.cs]: > "Password must have at least eight characters, at least one uppercase letter, one lowercase letter and one number"

This error string occurs in every validator with a password. Perhaps we should introduce string constants for common error messages

**Review by gingerbread-monster — CHANGES_REQUESTED** [review]

File: CharlieBackend.Api/Validators/AccountDTOValidators/ChangeCurrentPasswordDtoValidator.cs (hunk 2 above)
Comment (TetianaDomanska-dev) [reply CharlieBackend.Api/Validators/AccountDTOValidators/ChangeCurrentPasswordDtoValidator.cs]: It have already done in https://github.com/ita-social-projects/WhatBackend/pull/437

File: CharlieBackend.Api/Validators/AccountDTOValidators/ChangeCurrentPasswordDtoValidator.cs (hunk 3 above)
Comment (TetianaDomanska-dev) [reply CharlieBackend.Api/Validators/AccountDTOValidators/ChangeCurrentPasswordDtoValidator.cs]: Ok

File: CharlieBackend.Api/Validators/AccountDTOValidators/AuthenticationDtoValidator.cs (hunk 4 above)
Comment (TetianaDomanska-dev) [reply CharlieBackend.Api/Validators/AccountDTOValidators/AuthenticationDtoValidator.cs]: Done in https://github.com/ita-social-projects/WhatBackend/pull/435/commits/0c41ce8c05fe9bcf77febbc04a6fae5a668c4d6f

File: CharlieBackend.Business/Helpers/ValidationConstants.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+{
+    public static class ValidationConstants
+    {
+        public const int _minLength = 8;
+        public const int _maxLengthPassword = 30;
+        public const int _maxLengthEmail = 50;
+        public const int _maxLengthName = 30;
+        public const int _maxLengthURL = 200;
+        public const string _passwordRule = "Password must have at least eight characters, at least one uppercase letter, one lowercase letter and one number";
+        public const string _passwordConfirmNotValid = "Passwords do not match";
```
Comment (gingerbread-monster) [inline CharlieBackend.Business/Helpers/ValidationConstants.cs]: I think these constants should be named in PascalCase since they are `public`

Comment (TetianaDomanska-dev) [reply CharlieBackend.Business/Helpers/ValidationConstants.cs]: Ok, done https://github.com/ita-social-projects/WhatBackend/pull/435/commits/5a81642eeec62d7a428ce7447822ea8f613616ff

**Review by gingerbread-monster — APPROVED** [review]

**Review by zelya2021 — APPROVED** [review]

**Pull request merged**